examples/utils_rf.py

```diff
@@ -0,0 +1,131 @@
+"""
+===================================================================
+July 2020 --- Andrew Williams
+===================================================================
+Utility functions for plotting and analysis
+===================================================================
+"""
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+def add_121_line(ax):
+    import numpy as np
+    lims = [
+        np.min([ax.get_xlim(), ax.get_ylim()]),  # min of both axes
+        np.max([ax.get_xlim(), ax.get_ylim()]),  # max of both axes
+    ]
+
+    # now plot both limits against eachother
+    ax.plot(lims, lims, 'k-', alpha=0.75, zorder=0)
+    ax.set_xlim(lims)
+    ax.set_ylim(lims)
+    
+def prettify_plot(ax):
+    """utility function for making plots prettier"""
+    ax.get_xaxis().tick_bottom()
+    ax.get_yaxis().tick_left()
+    ax.spines['left'].set_position(('outward', 10))
+    ax.spines['bottom'].set_position(('outward', 10))
+
+    ax.spines['top'].set_visible(False)
+    ax.spines['right'].set_visible(False)
+    
+    
+def plot_results(ax, truth, pred, title):
+    from sklearn.metrics import median_absolute_error, r2_score, mean_squared_error
+    from sklearn.model_selection import train_test_split
+    ax.scatter(y=pred, x=truth, zorder=2.6, s=20, alpha=0.5)
+    props = {'boxstyle':'round', 'facecolor':'wheat'}
+    ax.text(x=0.05, y=0.8, transform=ax.transAxes,
+            s=f'$R^2$={np.round(r2_score(pred, truth),2)}, ' +\
+                f'MAE={np.round(median_absolute_error(pred, truth),3)}, ' +\
+                f'RMSE={np.round( np.sqrt(mean_squared_error(pred, truth)), 2)}',
+            bbox=props)
+
+    # Change axis formatting
+    prettify_plot(ax)
+    add_121_line(ax)
+    ax.set_title(title)
+    ax.set_xlabel('Truth')
+    ax.set_ylabel('Prediction')
+    
+    
+def LeaveOneOut(Xdata, Ydata, rndseed=0, **rf_kwargs):
```

Comment:
This is really cool! I think it could be really useful for the other models too though, would you mind making it take the model as a parameter to make it more general? You could also consider using the existing scikit-learn [LeaveOneOut](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.LeaveOneOut.html) interface :-)

Reply:
I did try the LeaveOneOut interface from scikit but had a bit of trouble using it for more complex array types... I've made this function more general though, it now takes a `model` argument and can also do a baseline linear regression - because why not? haha